conans/client/generators/virtualbuildenv.py

```diff
@@ -9,12 +10,13 @@ def __init__(self, conanfile):
         super(VirtualBuildEnvGenerator, self).__init__(conanfile)
         self.venv_name = "conanbuildenv"
         compiler = conanfile.settings.get_safe("compiler")
-        if compiler != "Visual Studio":
-            tools = AutoToolsBuildEnvironment(conanfile)
+        if compiler == "Visual Studio":
+            self.env = VisualStudioBuildEnvironment(conanfile).vars_dict
+            for key, value in vcvars_dict(conanfile.settings).items():
+                if not key.startswith("CONAN"):
```

Comment:
I think I understand why filtering the "CONAN" vars but it is a little arbitrary.
The `vcvars_dict` is returning ALL the read environment after applying an vcvars command, but here I assume we don't want to dump to the activate the whole environment. So, probably we should improve the `vcvars_dict` function (probably opt-in to not break) to return the only new/modified values in the environment. 
Let me take a look.

Reply:
https://github.com/conan-io/conan/pull/2948

Reply:
Update with develop. The default behaviour for `vcvars_dict` now is to only return the variables modified/created. So, does the `startswith("CONAN")` makes sense anymore?